src/esp/gfx/LightSetup.cpp

```diff
@@ -15,32 +15,63 @@ bool operator!=(const LightInfo& a, const LightInfo& b) {
   return !(a == b);
 }
 
-Magnum::Vector3 getLightPositionRelativeToCamera(
+Magnum::Vector4 getLightPositionRelativeToCamera(
     const LightInfo& light,
     const Magnum::Matrix4& transformationMatrix,
     const Magnum::Matrix4& cameraMatrix) {
+  ASSERT(light.position.w() == 1 || light.position.w() == 0);
+  const auto pos = light.position.xyz();
+
+  bool isDirectional = light.position.w() == 0;
+  Magnum::Vector4 out{0};
+
   switch (light.model) {
     case LightPositionModel::OBJECT:
-      return transformationMatrix.transformPoint(light.position);
+      out.xyz() = isDirectional ? transformationMatrix.transformVector(pos)
+                                : transformationMatrix.transformPoint(pos);
+      break;
```

Comment:
Since you have a four-component input already, there's no need for any branching or composing the return value out of single components:

```cpp
    case LightPositionModel::OBJECT:
      return transformationMatrix*light.position;
    case LightPositionModel::GLOBAL:
      return cameraMatrix*light.position;
    case LightPositionModel::CAMERA:
      return light.position;
```

That's basically what [transformVector()](https://doc.magnum.graphics/magnum/classMagnum_1_1Math_1_1Matrix4.html#a950213df0114f0eb91800ab4f79972e9) / [transformPoint()](https://doc.magnum.graphics/magnum/classMagnum_1_1Math_1_1Matrix4.html#ace937774e16d8e3a32518d23dcf3369c) do internally.